sensor_msgs/msg/NavSatStatus.msg

```diff
@@ -4,16 +4,18 @@
 # type and the last time differential corrections were received.  A
 # fix is valid when status >= STATUS_FIX.
 
+int8 STATUS_UNKNOWN = -2        # status is not yet set
 int8 STATUS_NO_FIX =  -1        # unable to fix position
 int8 STATUS_FIX =      0        # unaugmented fix
 int8 STATUS_SBAS_FIX = 1        # with satellite-based augmentation
 int8 STATUS_GBAS_FIX = 2        # with ground-based augmentation
 
-int8 status
+int8 status -2 # STATUS_UNKNOWN
 
 # Bits defining which Global Navigation Satellite System signals were
 # used by the receiver.
 
+uint16 SERVICE_UNKNOWN = 0
```

Comment:
Looking at this now, it doesn't seem to make sense to add this value. The service field is a logical OR of these constants. It doesn't make sense to ask e.g. `if (status & 0 != 0)` which is exactly what this constant would lead to. Unknown service would need to be represented differently.

I guess differentiating between "no service" and "unknown service" can be done by combining `service` with `status`. I.e. `service == 0` with fix means "service unknown", while `service == 0` without fix means no service. Just the combination of `status == 0` and unknown fix would be problematic, but in that case, I think service type is not a big concern.

I suggest removing this constant.

Reply:
I think that this is adequately differentiated from status. 

You can have a `[[S|G]BAS_]FIX` with an unknown service type. Especially if there's potentially one not covered by the below enumeration. Such as a custom one based on local fiducials or private infrastructure that's not in the enumeration. 

I think that it's important that the default value not be a specific value such that it's clear that it's uninitialized or else someone has chosen to keep it as unknown.

Reply:
I'm not sure I'm reading your response correctly and in a unique interpretation. Are you for removing the constant? Are you for changing the value of `SERVICE_UNKNOWN` to a non-zero? Or something else?

Reply:
I was suggesting leaving it as is proposed with `SERVICE_UNKNOWN = 0` as a constant. 

If you want to know the quality of the data you check status level.

If you have logic that's GPS specific you check `if(service & SERVICE_GPS)`

The value SERVICE_UNKNOWN=0 will never match a check of `service & SERVICE_TYPE` if you check all known services. Conversely if you want to check that it's a known service type but don't care which one you just check that it's greater than zero. 

If we don't have the constant here, we'll end up with zero in the value and then 0 is just undefined instead of explicitly knowing that it's unknown.

Reply:
Well, in a bitfield, I don't think it's necessary to specify the meaning of 0x0000. That just means none of the bits are set. Nothing more, nothing less. The interpretations may differ based on context (i.e. `status` in this case).

But I must agree it doesn't hurt too much to have the constant there. The worst thing that can happen is that someone won't check the integer value of `SERVICE_UNKNOWN`, will use the `(service & SERVICE_UNKNOWN)` check and will wonder why it doesn't work when all other values work properly with the same kind of check. But I guess this is a kind of programming intuition most programmers will get correct.

Reply:
What about adding this comment to prevent programmers doing the one bad thing that can be done with this constant?

```suggestion
uint16 SERVICE_UNKNOWN = 0  # Remember service is a bitfield, so checking (service & SERVICE_UNKNOWN) will not work. Use == instead.
```

Reply:
That looks like a good way to avoid that potential mistake.